spec/controllers/api/v1/zones_spec.rb

```diff
@@ -0,0 +1,150 @@
+require "rails_helper"
+
+RSpec.describe "Api::V1::Zones", type: :request do
+  let(:admin) { create(:user, :admin) }
+  let(:supervisor) { create(:user, :supervisor, admin: admin) }
+  let!(:zone) { create(:zone, user: supervisor) }
+
+  before do
+    allow_any_instance_of(Api::V1::BaseController)
+      .to receive(:current_user)
+      .and_return(supervisor)
+  end
+
+  def json_response
+    JSON.parse(response.body)
+  end
+
+  describe "GET /api/v1/zones" do
+    subject { get "/api/v1/zones", params: params }
```

Comment:
```suggestion
    subject { get :index, params: params }
```
type controller thì get sẽ là action
Fix tương tự cho các action khác